src/main/java/io/stargate/sgv2/jsonapi/api/model/command/builders/SortClauseBuilder.java

```diff
@@ -64,178 +57,67 @@ public SortClause build(JsonNode node) {
-        SortExpression exp = SortExpression.vectorizeSearch(vectorizeData);
-        sortExpressions.clear();
-        sortExpressions.add(exp);
-        // TODO: aaron 17-oct-2024 - this break seems unneeded as above it checks if there is only 1
-        // field, leaving for now
-        break;
-      } else if (inner.getValue().isArray()) {
-        // TODO: HACK: quick support for tables, if the value is an array we will assume the column
-        // is a vector then need to check on table pathway that the sort is correct.
-        // NOTE: does not check if there are more than one sort expression, the
-        // TableCqlSortClauseResolver will take care of that so we can get proper ApiExceptions
-        // this is also why we do not break the look here
-        sortExpressions.add(
-            SortExpression.tableVectorSort(path, arrayNodeToVector((ArrayNode) inner.getValue())));
-      } else if (inner.getValue().isTextual()) {
-        // TODO: HACK: quick support for tables, if the value is an text  we will assume the column
-        // is a vector and the user wants to do vectorize then need to check on table pathway that
-        // the sort is correct.
-        // NOTE: does not check if there are more than one sort expression, the
-        // TableSortClauseResolver will take care of that so we can get proper ApiExceptions
-        // this is also why we do not break the look here
-        sortExpressions.add(SortExpression.tableVectorizeSort(path, inner.getValue().textValue()));
-      } else {
-        if (!inner.getValue().isInt()
-            || !(inner.getValue().intValue() == 1 || inner.getValue().intValue() == -1)) {
-          throw ErrorCodeV1.INVALID_SORT_CLAUSE_VALUE.toApiException(
-              "Sort ordering value can only be `1` for ascending or `-1` for descending (not `%s`)",
-              inner.getValue());
-        }
-
-        boolean ascending = inner.getValue().intValue() == 1;
-        SortExpression exp = SortExpression.sort(validatedPath, ascending);
-        sortExpressions.add(exp);
-      }
+      sortExpressions.add(
+          buildAndValidateExpression(validateSortClausePath(path), inner.getValue(), totalFields));
     }
     return new SortClause(sortExpressions);
   }
 
-  private String validateSortClausePath(String path) {
-    if (!NamingRules.FIELD.apply(path)) {
-      if (path.isEmpty()) {
-        throw ErrorCodeV1.INVALID_SORT_CLAUSE_PATH.toApiException(
-            "path must be represented as a non-empty string");
+  protected abstract SortExpression buildAndValidateExpression(
+      String validatedPath, JsonNode innerValue, int totalFields);
+
+  protected SortExpression defaultBuildAndValidateExpression(
+      String validatedPath, JsonNode innerValue) {
+    if (!innerValue.isInt()) {
+      // Special checking for String and ArrayNode to give less confusing error messages
+      if (innerValue.isArray()) {
+        throw ErrorCodeV1.INVALID_SORT_CLAUSE_VALUE.toApiException(
```

Comment:
I'll let those old V1 errors pass, assuming we are lift-and-shift from previous logic

Reply:
Exactly. Validation logic will (need to) change, too, as will exceptions.